### DISCUSSION TITLE
Welcome to DSharpPlus.SlashCommands Discussions!

### DISCUSSION THREAD

**Discussion by Soyvolon** [Soyvolon/DSharpPlus.SlashCommands#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other bot developers. We hope that you:
  * Ask questions about this Library (its a WIP so please ask questions)
  * Share ideas, both for command usage and how we can improve!
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell let us know how you are using Slash Commands

**Comment by Soyvolon** [comment]
I'm Soyvolon and I started building this so I could use Slash Interactions in my code similarly to how Commands Next is used. I may have gone a bit overboard seeing as I only wanted this for testing a single command for a single bot, but hey, others can use this now too so that is worth it in my book.

**Comment by ghost** [comment]
Well, I'm Dreas and I am using the Slash Commands to set certain playful commands as slashable. 

1 question I do have, is if you would update it so you don't need the version number in the Command inorder to call it. I mainly ask this as its the only working Slash Command Package that I could find, and having to put an integer to use the command can make it confusing for users in my server who dont exactly know about it.

Other than that question, this is an awesome package man. Hope you keep adding to it.

**Reply by Soyvolon** [reply]
Heyo! The version in the `[SlashCommand]` attribute is for the Developer only. It is a not a part of any slash commands, and the only way something similar would show up for your slash commands is if you also put a similarly named argument in your command method definition.

Hope that clarifies some things for you!

**Reply by ghost** [reply]
Ah! Ye that does. Im still a little confused cause I'm using your example Gateway bot, and I keep getting an error saying my commands aren't present in a dictionary. I'm extremely confused.

**Reply by Soyvolon** [reply]
Mind sending me your command class for me? I'll see if I can help.

**Reply by ghost** [reply]
I got that fixed actually. But I get an error stating that 

> 'A Discord Client or Sharded Client is required.'
and ive followed both the Readme guide and the example.

**Reply by Soyvolon** [reply]
That means you aren't passing a `DiscordClient` or `DiscordShardedClient` into the configuration.

**Reply by ghost** [reply]
Ye im dumb, i had it above where i declared the client. But now for an actual non user error thingy, Everytime I try to run the bot, it says that my slash commands already exist, which they dont. Even when I rename them, it throws that error.

**Reply by Soyvolon** [reply]
What does the error specifically say?

**Reply by ghost** [reply]
Even if I rename the command `Ping` it throws this. 
`System.Exception: 'A command with the name Ping already exsists.'`
